tests/phpunit/tests/media.php

```diff
@@ -6445,6 +6445,56 @@ public function test_wp_img_tag_add_auto_sizes( string $input, string $expected
 		);
 	}
 
+	/**
+	 * @ticket 62305
+	 *
+	 * @dataProvider data_image_converted_to_other_format_has_correct_filename
+	 *
+	 * @param bool $apply_big_image_size_threshold True if filter needs to apply, otherwise false.
+	 */
+	public function test_image_converted_to_other_format_has_correct_filename( bool $apply_big_image_size_threshold ) {
+		$temp_dir      = get_temp_dir();
+		$file          = $temp_dir . '/test-image.heic';
+		$scaled_suffix = $apply_big_image_size_threshold ? '-scaled' : '';
+		copy( DIR_TESTDATA . '/images/test-image.heic', $file );
+
+		$editor = wp_get_image_editor( $file );
+
+		// Skip if the editor does not support HEIC.
+		if ( ! $editor->supports_mime_type( 'image/heic' ) ) {
```

Comment:
Seems this needs to be:
```suggestion
		if ( ! is_wp_error( $editor ) && $editor->supports_mime_type( 'image/webp' ) ) {
```

( `wp_get_image_editor( $file )` returns WP_Error if the image is not supported).